src/sync/generic_network_transport.hpp

```diff
@@ -109,13 +105,19 @@ enum class ServiceErrorCode {
     none = 0
 };
 
+const std::error_category& custom_error_category() noexcept;
+std::error_code make_custom_error_code(int code) noexcept;
+
 ServiceErrorCode service_error_code_from_string(const std::string& code);
 const std::error_category& service_error_category() noexcept;
 std::error_code make_error_code(ServiceErrorCode) noexcept;
 
 const std::error_category& http_error_category() noexcept;
 std::error_code make_http_error_code(int http_code) noexcept;
 
+const std::error_category& client_error_category() noexcept;
+std::error_code make_client_error_code(ClientErrorCode) noexcept;
+
```

Comment:
Thanks for adding these, can we get a `AppError::is_client_error()` as well (nice to have for SDK's)

Reply:
done 👍🏼